src/sentry/api/serializers/models/dashboard.py

```diff
@@ -143,11 +143,26 @@ def get_attrs(self, item_list, user):
         return result
 
     def serialize(self, obj, attrs, user, **kwargs):
+        page_filter_keys = ["environment", "range", "start", "end"]
```

Comment:
We should raise out of retention error if start/end is out of retention, something like [this](https://github.com/getsentry/sentry/blob/264bc5f4d369cd9a6bed2ff18ec52b50789eeba6/src/sentry/api/serializers/models/discoversavedquery.py#L44)

Reply:
Added!